### DISCUSSION TITLE
Zoom (crosshair selection) no longer working after defining "zoom limits"

### DISCUSSION THREAD

**Discussion by AleXSR700** [dbuezas/lovelace-plotly-graph-card#313] [Q&A]
Hello everyone,

I wanted to try defining zero as my lower limit for my charts and found that when I use [this solution](https://github.com/dbuezas/lovelace-plotly-graph-card/discussions/233), the zoom crosshair no longer appears, so I cannot zoom.

The mouse wheel zome gets reset immediately, too.

![Animation](https://github.com/dbuezas/lovelace-plotly-graph-card/assets/57034871/96c6960f-e165-424d-82a5-188a8a76a7af)

```
type: custom:plotly-graph
hours_to_show: current_week
entities:
  - entity: sensor.schenker_vision_16_pro_energy_power
  - entity: sensor.sony_kd_75xf9005_naim_audio_av_energy_power_0
    
layout:
  uirevision: $fn () => Math.random()
  yaxis:
    range: |
        $fn ({ getFromConfig }) => {
          const all = getFromConfig("entities").flatMap(({ y }) => y);
          return [
            Math.max(0, Math.min(...all))
          ]
        }
```

Can anybody spot the issue?

Thank you for your help
Alex

**Comment by dbuezas** [comment]
Hey Alex, the range attribute expects an array with two values: one for the minimum and one for the maximum. You are only passing one, which I assume is making plotly behave in weird ways.
Array is computer speak for roughly "a list of things", and in JavaScript is denoted by things separated by commas and surrounded by square brackets.
Try with: 
```
return [
            0,
            Math.max(0, Math.min(...all)),    
          ]
```

**Reply by AleXSR700** [reply]
Oh, I should not have missed that -.-
I think I have gotten used to HA frontend telling me that a code is wrong and disabling the entire card rather than stay useable and break one feature of the card.

Tried your code but was surprised when I saw that you defined a "0".
So, the result is that the range is now fixed from -1 to 1 and zooming is still disabled :)

Also tried
```
    range: |
        $fn ({ getFromConfig }) => {
          const all = getFromConfig("entities").flatMap(({ y }) => y);
          return [
            Math.max(0, Math.min(...all)), // cap to lowest value, but at least 0
            Math.min(Math.max(...all), Math.max(...all)), // cap to highest value, but at most highest value    
                ]
        }
```
but that also did not restore zooming with crosshair.

But even with your original code (limits 12 and 18), zooming does not work for me.

![Animation](https://github.com/dbuezas/lovelace-plotly-graph-card/assets/57034871/fe5117c0-fc90-449f-a779-a45c8a31259a)

**Reply by dbuezas** [reply]
I can unfortunately not reproduce this and I don't understand what could be going on there.
Btw, in the max range value you can just say 
`Math.max(...all)`

The minimum between the max and the max is still the max

**Reply by AleXSR700** [reply]
How strange.
Maybe I will uninstall all custom cards and try again to see if it could be an effect of a different card. But I don't really see how :(

**Reply by dbuezas** [reply]
Try explicitly setting fixed range to false:
```
layout:
  yaxis:
    fixedrange: false
  xaxis:
    fixedrange: false
```

**Reply by AleXSR700** [reply]
No, unfortunately no change :(
